src/xbot/common/controls/sensors/MockGamepad.java

```diff
@@ -1,40 +1,164 @@
 package xbot.common.controls.sensors;
 
+import xbot.common.math.XYPair;
+
 public class MockGamepad implements XGamepad {
 
-    MockJoystick mockLeftStick = new MockJoystick();
-    MockJoystick mockRightStick = new MockJoystick();
-    MockJoystick mockDpad = new MockJoystick();
-    MockJoystick mockLeftTrigger = new MockJoystick();
-    MockJoystick mockRightTrigger = new MockJoystick();
+    XYPair leftJoystickAxis;
+    XYPair rightJoystickAxis;
+    double leftTriggerAxis;
+    double rightTriggerAxis;
+    boolean leftBumper;
+    boolean rightBumper;
+    boolean AButton;
+    boolean BButton;
+    boolean XButton;
+    boolean YButton;
+    boolean backButton;
+    boolean startButton;
+    boolean leftStickButton;
+    boolean rightStickButton;
+    
+    public void setLeftStick(double x, double y){
+        leftJoystickAxis = new XYPair(x, y);
+    }
+    
+    public void setRightStick(double x, double y){
+        rightJoystickAxis = new XYPair(x, y);
+    }
+    
+    public void setLeftTrigger(double x){
+        leftTriggerAxis = x;
+    }
+    
+    public void setRightTrigger(double x){
+        rightTriggerAxis = x;
+    }
+    
+    /**
+     * Sets the bumper state. True means pressed/on. False means not pressed/off.
```

Comment:
I'm moderately annoyed that this comment is on one of these methods but not the other :laughing: